### PR TITLE
test(cirrus): Cirrus preview flag

### LINKED ISSUE #11103: Add new integration test to support QA mode
We added support on cirrus to support preview mode, and experimenter now pushes to web preview collection, add some integration to test the qa mode flag

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4665)

### PR DESCRIPTION
**Pull request by yashikakhurana**
Because

- We added preview flag support for Cirrus to enable them to QA

This commit

- Add a new test to test preview flag functionality

Fixes #11103 
Note: Working on to send it to main collection

### CODE REVIEW

File: experimenter/tests/integration/nimbus/test_cirrus_integration.py [hunk 1]
```diff
@@ -247,3 +247,48 @@ def test_check_cirrus_targeting(
...
+    result_text_element = demo_app.wait_for_result_text(["Not Enrolled"])
+    assert result_text_element.is_displayed()
+    # enable nimbus preview flag
+    demo_app.fill_and_send_form_data("test", '{"test1":"test2"}', nimbus_preview=True)
+    demo_app.click_send_my_details()
+
+    displayed_text = demo_app.wait_for_result_text(
+        ["Control branch", "Treatment branch"]
+    ).text
+    assert displayed_text in ["Control branch", "Treatment branch"]
```
Comment (jaredlockhart) [inline experimenter/tests/integration/nimbus/test_cirrus_integration.py:283]: Pull all these references to `"Control branch"` `"Treatment branch"` out into constants pls 🙏

File: experimenter/tests/integration/nimbus/test_cirrus_integration.py [hunk 2]
```diff
@@ -247,3 +247,48 @@ def test_check_cirrus_targeting(
...
+    navigate_to(selenium)
+    selenium.refresh()
+    result_text_element = demo_app.wait_for_result_text(["Not Enrolled"])
+    assert result_text_element.is_displayed()
+    # not using nimbus preview flag
+    demo_app.fill_and_send_form_data("example1", '{"test1":"test2"}')
+    demo_app.click_send_my_details()
+
+    result_text_element = demo_app.wait_for_result_text(["wicked"])
+    assert result_text_element.is_displayed()
```
Comment (jaredlockhart) [inline experimenter/tests/integration/nimbus/test_cirrus_integration.py:294]: There's a lot of string constants and methods for querying the demo app we should be pulling out into constants on the demo app page object.

File: experimenter/tests/integration/nimbus/pages/demo_app/frontend.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ def wait_for_result_text(self, text_list):
 
         return self.wait_for_and_find_element(By.XPATH, xpath, description=None)
 
-    def fill_and_send_form_data(self, client_id, context):
+    def fill_and_send_form_data(self, client_id, context, nimbus_preview=False):
```
Comment (jaredlockhart) [inline experimenter/tests/integration/nimbus/pages/demo_app/frontend.py:17]: We have this DemoAppPage class here, can we move all the string constants onto constants here with some convenience methods for interacting with this page so the tests aren't quite so long? 🙏

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
Hooray!!  Couple ideas to clean this up a bit 🙏

**Review by jaredlockhart — APPROVED** [review]
Awesome looks great @yashikakhurana 🎉 🎉 🎉 🎉

**Comment by b4handjr** [comment]
I didn't look at this but we talked at MozWeek, nice job here!

**Pull request merged**